src/Adapters/Storage/Mongo/MongoStorageAdapter.js

```diff
@@ -687,13 +689,10 @@ export class MongoStorageAdapter implements StorageAdapter {
     };
 
     return this._adaptiveCollection(className)
-      .then(
-        collection =>
-          new Promise((resolve, reject) =>
-            collection._mongoCollection.createIndex(indexCreationRequest, indexOptions, error =>
-              error ? reject(error) : resolve()
-            )
-          )
+      .then(collection =>
+        collection._mongoCollection
+          .createIndex(indexCreationRequest, indexOptions)
+          .catch(error => this.handleError(error))
```

Comment:
Why ` .catch(error => this.handleError(error))` if there is already a `handleError` after this?

Reply:
Yes you right :), i revert this change :)